### PR TITLE
Support for running Avitab as standalone add-on to Microsoft Flight Simulator 2020

### PR DESCRIPTION
**Pull request by mjh65**
This update supports the build of a new Avitab standalone variant that can be run alongside Microsoft Flight Simulator 2020.
There are currently many restrictions, but it may be useful for some small subset of VR flyers.
The new executable will only be created if the MSFS SDK is installed on the build machine. Env. var. MSFS_SDK msut be set.
The new executable [Avitab-msfs.exe] still uses the X-Plane installation to read NAV data, so it's only really useful for VR flyers who have both sims installed.
The Map app will centre on the user aircraft current location. Any other aircraft that are shown will be simulated AI aircraft, not other simulators in a private group, or in the worldwide public gamesphere. This is a known and reported defect of the MSFS SDK.
When used in VR the Avitab-msfs window will need to be 'imported' into the VR space, eg using Steam's desktop window overlay support, and some jiggery-pokery will be necessary to switch focus between MSFS and Avitab to target mouse/VR controllers as required.
In summary, this is a very early proof-of-concept update, but perhaps still worth wrapping into the main Avitab repo?

### CODE REVIEW

File: src/environment/msfs/MsfsAddonEnvironment.cpp [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    } else {
+        return otherLocations[id-1];
+    }
+
+    return loc;
+}
+
+void MsfsAddonEnvironment::setZoomLevel(int zoom)
+{
+    // for zoom levels up to 8 request max range of 200km
```
Comment (fpw) [inline src/environment/msfs/MsfsAddonEnvironment.cpp:88]: I think the meaning of the zoom level depends on the map type that's being used. For EPSG tiles, the meaning is fixed. But when using charts as a map, the meaning changes. Maybe `OverlayedMap::getMapWidthNM` is easier to deal with than the raw zoom level.

Comment (mjh65) [reply src/environment/msfs/MsfsAddonEnvironment.cpp]: Have discussed this with David, and agreed that knowing the current range of the map isn't really important, and better to just request aircraft in a fixed range. So we have removed the changes to the zoom level APIs.

File: src/environment/msfs/MsfsAddonEnvironment.cpp [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+}
+
+AircraftID MsfsAddonEnvironment::getActiveAircraftCount()
+{
+    return 1 + otherLocations.size();
+}
+
+Location MsfsAddonEnvironment::getAircraftLocation(AircraftID id)
+{
+    Location loc;
```
[the remark this answers is not in the document]
Comment (mjh65) [reply src/environment/msfs/MsfsAddonEnvironment.cpp]: David has checked this, and added mutexes matching those in the XPlane environment.

**Comment by mjh65** [comment]
Revised PR for your consideration. I have been rebasing this branch on your latest master, so the 2 commits of this PR represent the original request and then the modifications to address your comments. Thanks to dave6502 for picking this up and pushing it forward.

**Review by fpw — APPROVED** [review]

**Comment by fpw** [comment]
Thanks for revising the PR! GitHub shows that there are merge conflicts so I can't merge it yet, I think it's mainly because the line endings are different on your side, maybe git's autocrlf setting is different on your machine. Could you please change the line ending style to match that of the master, i.e. so that no changes are shown in the PR diff for MapApp.cpp as an example?
